apps/cowswap-frontend/src/modules/tradeFlow/hooks/useSafeBundleFlowContext.ts

```diff
@@ -7,29 +7,36 @@ import useSWR from 'swr'
 
 import { useReceiveAmountInfo } from 'modules/trade'
 
-import { useGP2SettlementContract, useTokenContract, useWETHContract } from 'common/hooks/useContract'
+import { useGP2SettlementContract, useTokenContract, useWethContract } from 'common/hooks/useContract'
 import { useNeedsApproval } from 'common/hooks/useNeedsApproval'
 import { useTradeSpenderAddress } from 'common/hooks/useTradeSpenderAddress'
 
 import { SafeBundleFlowContext } from '../types/TradeFlowContext'
 
 export function useSafeBundleFlowContext(): SafeBundleFlowContext | null {
-  const settlementContract = useGP2SettlementContract()
+  const { contract: settlementContract, chainId: settlementChainId } = useGP2SettlementContract()
   const spender = useTradeSpenderAddress()
 
   const safeAppsSdk = useSafeAppsSdk()
-  const wrappedNativeContract = useWETHContract()
+  const { contract: wrappedNativeContract, chainId: wrappedNativeChainId } = useWethContract()
+
   const receiveAmountInfo = useReceiveAmountInfo()
   const inputAmountWithSlippage = receiveAmountInfo?.afterSlippage.sellAmount
   const needsApproval = useNeedsApproval(inputAmountWithSlippage)
   const inputCurrencyAddress = useMemo(() => {
     return inputAmountWithSlippage ? getCurrencyAddress(inputAmountWithSlippage.currency) : undefined
   }, [inputAmountWithSlippage])
-  const erc20Contract = useTokenContract(inputCurrencyAddress)
+  const { contract: erc20Contract, chainId: erc20ChainId } = useTokenContract(inputCurrencyAddress)
 
   return (
     useSWR(
-      settlementContract && spender && safeAppsSdk && wrappedNativeContract && erc20Contract
+      settlementChainId === erc20ChainId &&
+        settlementChainId === wrappedNativeChainId &&
+        settlementContract &&
+        spender &&
+        safeAppsSdk &&
+        wrappedNativeContract &&
+        erc20Contract
```

Comment:
New variable for this?

Reply:
not sure if I got you, but you want me to do this? 5569899273ec580ec28924a59aef89c6d5d94c4b

Reply:
I can't tell, probably you haven't pushed your local changes